pkg/channel/distributed/common/kafka/sarama/sarama.go

```diff
@@ -308,73 +173,64 @@ func extractRootCerts(saramaConfigYamlString string) (string, *x509.CertPool, er
 	}
 
 	// Remove The RootPEMs From The Sarama YAML String (Multi-Line / Greedy To Collect All PEMs)
-	regex, err := regexp.Compile("(?s)\\s*RootPEMs:.*-----END CERTIFICATE-----")
-	if err != nil {
-		return saramaConfigYamlString, nil, err
-	} else {
-		updatedSaramaConfigYamlBytes := regex.ReplaceAll([]byte(saramaConfigYamlString), []byte{})
-		return string(updatedSaramaConfigYamlBytes), certPool, nil
-	}
+	regex := regexp.MustCompile(`(?s)\s*RootPEMs:.*-----END CERTIFICATE-----`)
+	updatedSaramaConfigYamlBytes := regex.ReplaceAll([]byte(saramaConfigYamlString), []byte{})
+	return string(updatedSaramaConfigYamlBytes), certPool, nil
 }
```

Comment:
So, you're changing the behavior here - it used to continue on if there was a problem with the Regexp but now it will panic.   Is that intentional?  I think what you have is ok in this case because the regexp is static.  We should. probably move it up to the top as a package private variable so that we're not creating the regexp every time through?

Reply:
This was one of the linter suggestions ("use MustCompile when the string is constant"); I'll move it to the top as suggested though, that makes sense.

Reply:
yeah, i think it's a good change - thanks!